.gitlab-ci.yml

```diff
@@ -327,3 +327,110 @@ publish:docker-image:automatic:
   rules:
     - if: '$CI_COMMIT_BRANCH =~ /^(master|staging|production|feature-.+|[0-9]+\.[0-9]+\.([0-9]+|x))$/'
   extends: .template:publish:docker-image
+
+####################
+# Hardware testing #
+####################
+
+test:hardware:build:
+  <<: *convert_raspbian
+  variables:
+    RASPBERRYPI_PLATFORM: raspberrypi4
+  when: manual
+  stage: build
+  script:
+    # Create the custom needed configurations
+    - |
+      cat <<-EOF > configs/hardware-testing-conf
+      MENDER_COMPRESS_DISK_IMAGE="none"
+      MENDER_ARTIFACT_COMPRESSION="none"
+      # Install the mender-client
+      MENDER_CLIENT_INSTALL="y"
+      # Enable sshd
+      function enable_sshd () {
+              log_warn "Enabling sshd by default"
+              touch work/boot/ssh
+      }
+      PLATFORM_MODIFY_HOOKS+=(enable_sshd)
+      # Allow root ssh logins
+      function allow_root_ssh_logins() {
+              log_warn "Allowing ssh root logins"
+              echo "PermitRootLogin yes" >> /etc/ssh/sshd_config
+      }
+      PLATFORM_MODIFY_HOOKS+=(allow_root_ssh_logins)
```

Comment:
I take it this is the reason we can not use the existing one?

Reply:
Yes